engine/v2/transaction/outlines/testabilities/txgraph/graph_builder.go

```diff
@@ -0,0 +1,179 @@
+	BEEFHex *string          // Hexadecimal representation of the transaction in BEEF format.
+	RawHex  *string          // Optional raw hexadecimal representation of the transaction.
+}
+
+// IsZero checks if the TransactionDescriptor is uninitialized.
+func (t TransactionDescriptor) IsZero() bool {
+	return t == TransactionDescriptor{}
+}
+
+// IsBeef checks if the actual transaction was serialized to HexBEEF format.
+func (t TransactionDescriptor) IsBEEF() bool { return t.BEEFHex != nil }
+
+// IsRaw checks if the actual transaction was serialized to Hex format.
+func (t TransactionDescriptor) IsRaw() bool { return t.RawHex != nil }
+
+// GraphBuilderTransactions is a map where the key is a transaction name (string)
+// and the value is a TransactionDescriptor containing details of the transaction.
+type GraphBuilderTransactions map[string]*TransactionDescriptor
+
+// AddRawTx adds a raw transaction to the GraphBuilderTransactions map.
+// The name is used as the key, and the transaction is stored with its hexadecimal representation.
+func (g GraphBuilderTransactions) AddRawTx(t *testing.T, name string, tx *sdk.Transaction) {
+	t.Helper()
+	require.NotEmpty(t, name, "Name of the raw transaction should not be empty")
+	require.NotNil(t, tx, "Added transaction should not be nil")
+
+	hex := tx.Hex()
+	g[name] = &TransactionDescriptor{Tx: tx, RawHex: &hex}
+}
+
+// AddMinedTx adds a mined transaction to the GraphBuilderTransactions map.
+// The name is used as the key, and the transaction is stored with its BEEF hexadecimal representation.
+func (g GraphBuilderTransactions) AddMinedTx(t *testing.T, name string, tx *sdk.Transaction) {
+	t.Helper()
+	require.NotEmpty(t, name, "Name of the mined transaction should not be empty")
+	require.NotNil(t, tx, "Added transaction should not be nil")
+
+	hex, err := tx.BEEFHex()
+	require.NoError(t, err, "Failed to generate BEEF hexadecimal for transaction")
+
+	g[name] = &TransactionDescriptor{Tx: tx, BEEFHex: &hex}
+}
+
+// GraphBuilder is a utility for constructing and managing Bitcoin transactions.
+// It supports creating mined and raw transactions, verifying scripts, and managing state.
+type GraphBuilder struct {
+	t                    *testing.T               // Test context for handling errors.
+	transactions         GraphBuilderTransactions // Collection of transactions managed by the builder.
+	p2pKHLockingScript   *script.Script           // Locking script for P2PKH transactions.
+	p2pKHUnlockingScript *p2pkh.P2PKH             // Unlocking script for P2PKH transactions.
+	hexGen               HexGen                   // Generator for unique hexadecimal values.
+	blockHeight          BlockHeight              // Current block height.
+	satoshis             uint64                   // Default amount of satoshis for mined transactions.
+}
+
+// GraphBuilderTransactions returns the map of transactions managed by the builder.
+func (g *GraphBuilder) GraphBuilderTransactions() GraphBuilderTransactions {
```

Comment:
```suggestion
func (g *GraphBuilder) Transactions() GraphBuilderTransactions {
```

because the call looks redundant, e.g.: `graphBuilder.GraphBuilderTransactions()`